client/src/components/Button/IconHOC.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+/**
+ * Higher order component that adds a `icon` props to another component and translate
+ * it to a matching class.
+ */
+const IconHOC = (Component) => {
```

Comment:
I made this an HOC component thinking we might end up reusing it on other components that need to display icons.

But it might be a bit overkill.

Reply:
This seems sensible. Are there any other examples where this would work currently?